### PR TITLE
Added new parameter to the Set-AzDataFactoryV2 cmdlet to supporet Rep…

### PR DESCRIPTION
**Pull request by sanjkuma**
…oConfiguration related settings. Added Support for -InputObject parameter and Piping of object. Added 5 parametersets. Added examples and ResourceIdCompleter.

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [X] For public API changes to cmdlets:
    - [X] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [X] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
Corresponding design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/259

**Comment by hvermis** [comment]
@cormacpayne @vladimir-shcherbakov We were hoping to have this PR in the release that is going out this week. The code has been ready since last week - can you merge it?

File: src/DataFactory/DataFactoryV2/Changelog.md [hunk 1]
```diff
@@ -20,6 +20,10 @@
 
 ## Upcoming Release
 
+## Version 1.0.3
```
Comment (cormacpayne) [inline src/DataFactory/DataFactoryV2/Changelog.md]: @sanjkuma please remove this header and place the below text under the `Upcoming Release` header

File: src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs [hunk 2]
```diff
@@ -14,54 +14,502 @@
...
+            ParameterSetName = ParameterSetNames.ByResourceIdFactoryRepoGitConfig,
+            Mandatory = true,
+            HelpMessage = Constants.HelpRepoConfigurationHostName)]
+        [ValidateNotNullOrEmpty]
+        #endregion // Attributes
+        public string HostName { get; set; }
+
         public override void ExecuteCmdlet()
         {
+            this.ByInputObject();
```
Comment (cormacpayne) [inline src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs]: @sanjkuma minor: can you create a new method `ValidateParameters()` that then calls these two methods, or even moves the contents of the two methods into this single method?

File: src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md [hunk 3]
```diff
@@ -134,6 +414,21 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -TenantId
+The tags of the data factory.
```
Comment (cormacpayne) [inline src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md]: @sanjkuma this description is incorrect -- what is the tenant id being passed to this cmdlet?

File: src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md [hunk 4]
```diff
@@ -119,12 +345,66 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -ResourceId
+The Azure resource ID.
```
Comment (cormacpayne) [inline src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md]: @sanjkuma minor: this should say what it's the resource id of (_e.g._, "The Azure resource id of the V2 data factory")

File: src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md [hunk 5]
```diff
@@ -74,12 +201,57 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -HostName
+The tags of the data factory.
```
Comment (cormacpayne) [inline src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md]: @sanjkuma this description needs to be updated

File: src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md [hunk 6]
```diff
@@ -38,12 +99,78 @@ PS C:\> Set-AzDataFactoryV2 -ResourceGroupName "ADF" -Name "WikiADF" -Location "
     Tags              : {}
     Identity          : Microsoft.Azure.Management.DataFactory.Models.FactoryIdentity
     ProvisioningState : Succeeded
+    RepoConfiguration :
+
+PS C:\> Get-AzDataFactoryV2 -ResourceGroupName "ADF" -Name "WikiADF" | Set-AzDataFactoryV2 -AccountName msdata -RepositoryName ADFRepo -CollaborationBranch master -RootFolder / -ProjectName "Azure Data Factory"
```
Comment (cormacpayne) [inline src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md]: @cormacpayne this example you added should be in it's own example, so you'll need to add another example header for this piping scenario

File: src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs (hunk 2 above)
Comment (sanjkuma) [reply src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs]: Done

File: src/DataFactory/DataFactoryV2/Changelog.md (hunk 1 above)
Comment (sanjkuma) [reply src/DataFactory/DataFactoryV2/Changelog.md]: Done.

File: src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md (hunk 3 above)
Comment (sanjkuma) [reply src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md]: Done

File: src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md (hunk 5 above)
Comment (sanjkuma) [reply src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md]: Done

File: src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md (hunk 6 above)
Comment (sanjkuma) [reply src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md]: Done.

File: src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md (hunk 4 above)
Comment (sanjkuma) [reply src/DataFactory/DataFactoryV2/help/Set-AzDataFactoryV2.md]: Done.

**Pull request merged**